### PR TITLE
BSM-56: Create/Update/Delete Business Services

### PR DESCRIPTION
**Pull request by DerTak**
Jira: http://issues.opennms.org/browse/BSM-56
Bamboo: http://bamboo.internal.opennms.com:8085/browse/OPENNMS-ONMS263

### CODE REVIEW

File: smoke-test/src/test/java/org/opennms/smoketest/BSMAdminIT.java [hunk 1]
```diff
@@ -28,70 +28,49 @@
...
-
-        // Verify that the Business Service appears in the list
-        m_driver.get(BASE_URL + "opennms/admin/bsm/index.jsp");
-        findElementByXpath("//*[contains(text(), '" + servicePrefix + "-name')]");
-
-        // The Business Service's attributes should also be displayed
-        WebElement el = findElementByXpath("//*[contains(text(), '" + servicePrefix + "-key')]");
-        WebElement parent = el.findElement(By.xpath(".."));
-        assertEquals(servicePrefix + "-key: " + servicePrefix + "-value", parent.getText());
+    @Test
```
Comment (j-white) [inline smoke-test/src/test/java/org/opennms/smoketest/BSMAdminIT.java]: These tests assume that they are executed in the order defined, but JUnit does not provide this guarantee.

We should either rework the logic or rename the tests and use the @FixMethodOrder annotation.

File: opennms-webapp/src/main/webapp/admin/bsm/js/controllers/BusinessServicesListController.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+            $scope.bsCreate = function () {
+                if ($scope.name.length > 0) {
+                    $scope.newBS = new BusinessServices({name: $scope.name}).$save()
+                        .then(function (newBS) {
+                            listBS();
+                            $scope.name = "";
+                        })
+                        .catch(function (req) {
+                            $log.warn("create for new BusinessService failed: " + req);
```
Comment (mvrueden) [inline opennms-webapp/src/main/webapp/admin/bsm/js/controllers/BusinessServicesListController.js]: please remove the `for`

File: smoke-test/src/test/java/org/opennms/smoketest/BSMAdminIT.java [hunk 3]
```diff
@@ -28,70 +28,49 @@
...
 
 public class BSMAdminIT extends OpenNMSSeleniumTestCase {
 
-    @Test
-    public void canReadBusinessServices() throws ClientProtocolException, IOException {
-        // Deferring this check until we can delete entries via the UI
-//        // Verify that a message is presented when there are no business services defined
-//        m_driver.get(BASE_URL + "opennms/admin/bsm/index.jsp");
-//        findElementByXpath("//*[contains(text(), 'There are no business services defined.')]");
+    public static final String RENAMED_SERVICE_NAME = "renamed_service";
```
Comment (mvrueden) [inline smoke-test/src/test/java/org/opennms/smoketest/BSMAdminIT.java]: Why is this declared public and everything else private?

File: smoke-test/src/test/java/org/opennms/smoketest/BSMAdminIT.java (hunk 1 above)
Comment (mvrueden) [inline smoke-test/src/test/java/org/opennms/smoketest/BSMAdminIT.java]: Please use @FixMethodOrder

File: smoke-test/src/test/java/org/opennms/smoketest/BSMAdminIT.java [hunk 4]
```diff
@@ -28,70 +28,49 @@
...
+        findElementById("delete-" + BASIC_SERVICE_NAME).click();
+        findElementByXpath("//*[contains(text(), 'There are no business services defined.')]");
+    }
 
-            try (CloseableHttpResponse response = httpClient.execute(request)) {
-                assertEquals(201, response.getStatusLine().getStatusCode());
-            }
-        }
+    @Test
+    public void canEditBusinessServices() {
```
Comment (mvrueden) [inline smoke-test/src/test/java/org/opennms/smoketest/BSMAdminIT.java]: Can you please seperate the code segments and add a comment to each of the segments what it is doing?
E.g.:

    //create new service
    ...
    // rename created service
    ...
    // delete created service
    ...

**Comment by j-white** [comment]
I tried the build and it works well.

File: smoke-test/src/test/java/org/opennms/smoketest/BSMAdminIT.java (hunk 1 above)
Comment (DerTak) [inline smoke-test/src/test/java/org/opennms/smoketest/BSMAdminIT.java]: fixed

**Pull request merged**